internal/kgateway/krtcollections/policy.go

```diff
@@ -57,6 +57,15 @@ func (e *BackendPortNotAllowedError) Error() string {
 	return fmt.Sprintf("BackendRef to \"%s\" includes a port. Do not specify a port when referencing a Backend resource, as it defines its own port configuration", e.BackendName)
 }
 
+type BackendPortNotFoundError struct {
+	Port        int32
+	BackendName string
+}
+
+func (b *BackendPortNotFoundError) Error() string {
+	return fmt.Sprintf("port %v of Service %q is not defined", b.Port, b.BackendName)
```

Comment:
```suggestion
	return fmt.Sprintf("port %v of Backend %q is not defined", b.Port, b.BackendName)
```

Doesn't have to be a k8s service